mne/preprocessing/tests/test_ica.py

```diff
@@ -143,22 +150,57 @@ def test_ica_simple(method):
     data = np.dot(A, S)
     ica = ICA(n_components=n_components, method=method, random_state=0)
     ica._fit(data, 0)
-    transform = np.dot(np.dot(ica.unmixing_matrix_, ica.pca_components_), A)
+    transform = ica.unmixing_matrix_ @ ica.pca_components_ @ A
     amari_distance = np.mean(np.sum(np.abs(transform), axis=1) /
                              np.max(np.abs(transform), axis=1) - 1.)
     assert amari_distance < 0.1
 
 
+@requires_sklearn
+@pytest.mark.parametrize('n_pca_components', (8, 9, 10))
+@pytest.mark.parametrize('max_pca_components', [
+    9,
+    pytest.param(0.999999, marks=pytest.mark.xfail(reason='XXX should work')),
```

Comment:
With 10 channels and an avg ref proj this gives 8 components instead of 9; if we use 1. instead, it works sometimes and not others because of numerical precision issues in the PCA.

If we want this to work (and I think we do) then we should change the logic to be "greater than or equal to" the given variance instead of "equal to". I consider this a bugfix. See previous [discussion](https://github.com/mne-tools/mne-python/pull/8321#discussion_r498159641).

Reply:
Our current logic is "less than or equal to" :)

Reply:
But I agree we should change it. That must be the reason why scikit-learn does it that way too.